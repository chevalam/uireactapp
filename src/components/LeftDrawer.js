import React, { Component } from 'react';
import List, {ListItem} from 'material-ui/List';
import AccountCircleIcon from 'material-ui-icons/AccountCircle';
import StarIcon from 'material-ui-icons/Star';
import VerifiedUserIcon from 'material-ui-icons/VerifiedUser';
import PeopleIcon from 'material-ui-icons/People';
import InboxIcon from 'material-ui-icons/Inbox';

class LeftDrawer extends Component {
  render() {
    return (
      <div>
    <List>
        <ListItem
          key={1}
          primaryText="All Users"
          leftIcon={<AccountCircleIcon />}
        />
        <ListItem
          key={2}
          primaryText="Favourites"
          leftIcon={<StarIcon />}
        />
        <ListItem
          key={3}
          primaryText="Administrators"
          leftIcon={<VerifiedUserIcon />}
        />
        <ListItem
          key={4}
          primaryText="Non-Admins"
          leftIcon={<PeopleIcon />}
        />
        <ListItem
          key={5}
          primaryText="Archived"
          leftIcon={<InboxIcon />}
        />
    </List>
    </div>
  )
  }
}
export default LeftDrawer;
