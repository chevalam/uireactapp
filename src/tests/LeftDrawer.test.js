import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import LeftDrawer from '../components/LeftDrawer';

configure({ adapter: new Adapter() });

describe('LeftDrawer', () => {
    it('should render correctly', () => {
        const output = shallow(
            <LeftDrawer />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
