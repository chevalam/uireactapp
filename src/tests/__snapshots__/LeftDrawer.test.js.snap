// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftDrawer should render correctly 1`] = `
<div>
  <List>
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="1"
      leftIcon={<pure(AccountCircle) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="All Users"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="2"
      leftIcon={<pure(Star) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Favourites"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="3"
      leftIcon={<pure(VerifiedUser) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Administrators"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="4"
      leftIcon={<pure(People) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Non-Admins"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement="span"
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="5"
      leftIcon={<pure(Inbox) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Archived"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
  </List>
</div>
`;
